### PR TITLE
lakectl local: add git context tracking

### PR DESCRIPTION
**Pull request by N-o-Z**

### CODE REVIEW

File: cmd/lakectl/cmd/local.go [hunk 1]
```diff
@@ -131,6 +132,16 @@ func warnOnCaseInsensitiveDirectory(path string) {
...
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		preRunCmd(cmd)
+
+		_, localPath := getSyncArgs(args, false, false)
+		cmdSuffix := ""
+		if git.IsRepository(localPath) {
+			cmdSuffix = "git"
+		}
+		sendStats(cmd, cmdSuffix)
+	},
```
Comment (guy-har) [inline cmd/lakectl/cmd/local.go:144]: 🤩

**Review by guy-har — CHANGES_REQUESTED** [review]
Looks good, 
some nit picking. only reason to request changes is the late version.IsVersionUnreleased()

File: cmd/lakectl/cmd/root.go [hunk 2]
```diff
@@ -409,29 +361,83 @@ var excludeStatsCmds = []string{
...
+		if cfgFile != "" {
+			// specific message in case the file isn't found
+			DieFmt("config file not found, please run \"lakectl config\" to create one\n%s\n", cfgErr)
+		}
+	// if the config file wasn't provided, try to run using the default values + env vars
+	case cfgErr != nil:
+		// other errors while reading the config file
+		DieFmt("error reading configuration file: %v", cfgErr)
+	}
+
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:393]: And there's a bug there 😄

File: cmd/lakectl/cmd/root.go [hunk 3]
```diff
@@ -409,29 +361,83 @@ var excludeStatsCmds = []string{
...
+				errStr = resp.Status()
+			}
+			if errStr != "" {
+				_, _ = fmt.Fprintf(os.Stderr, "Warning: failed sending statistics: %s\n", errStr)
+			}
+		}
+	}
+	if version.IsVersionUnreleased() {
+		return
 	}
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/root.go:436]: Slipped

**Review by guy-har — APPROVED** [review]
LGTM

**Pull request merged**